src/main/java/core/basesyntax/service/FruitTransactionValidator.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.service;
+
+public interface FruitTransactionValidator<T, U> {
+    void validateOperation(T t);
+
+    void validateFruit(String fruit);
+
+    void validateAmount(int amount,U u);
```

Comment:
The interface methods seem to be too specific for a generic 'FruitTransactionValidator'. The method names suggest that this validator is tailored to validate operations, fruits, and amounts. However, the interface itself is generic, which might be confusing. It would be better to either make the interface and its methods more generic or to rename the interface to reflect its specific purpose.